src/Uno.Toolkit.UI/Behaviors/SelectorExtensions.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Input;
+using Microsoft.Extensions.Logging;
+using Uno.Extensions;
+using Uno.Logging;
+using Windows.System;
+using Windows.UI.ViewManagement;
+using Windows.Foundation.Collections;
+using Uno.Disposables;
+
+#if IS_WINUI
+using Microsoft.UI.Xaml.Data;
+using Microsoft.UI.Xaml.Controls.Primitives;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Input;
+
+#else
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Input;
+using PipsPager = Microsoft.UI.Xaml.Controls.PipsPager;
+#endif
+
+namespace Uno.Toolkit.UI;
+
+/// <summary>
+/// Extensions for <see cref="Selector"/>
+/// </summary>
+[Bindable]
+public static partial class SelectorExtensions
+{
+
+	/// <summary>
+	/// Backing property for the <see cref="PipsPager"/> that will be linked to the desired <see cref="Selector"/> control.
+	/// </summary>
+	#region DependencyProperty: PipsPager
+	public static DependencyProperty PipsPagerProperty { get; } =
+	DependencyProperty.RegisterAttached("PipsPager", typeof(PipsPager), typeof(SelectorExtensions), new PropertyMetadata(null, OnPipsPagerChanged));
+
+	public static void SetPipsPager(Selector element, PipsPager value) =>
+		element.SetValue(PipsPagerProperty, value);
+
+	public static PipsPager GetPipsPager(Selector element) =>
+		(PipsPager)element.GetValue(PipsPagerProperty);
+	#endregion
+
+	static void OnPipsPagerChanged(DependencyObject dependencyObject, DependencyPropertyChangedEventArgs args)
+	{
+		if (args.NewValue == args.OldValue || dependencyObject is not Selector selector || args.NewValue is not PipsPager pipsPager)
```

Comment:
How come we clear the binding if NewValue == OldValue?

Reply:
If `old == new` we don't need to clean anything, because the state is the same, no?

Reply:
yeah but looking at this if statement, if old == new, then the if's condition will be `true` and we will run the code at lines 58-59 that will clear the binding